core/templates/pages/contributor-dashboard-page/modal-templates/certificate-download-modal.component.ts

```diff
@@ -0,0 +1,299 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the certificate download modal.
+ */
+
+import { Component, Input } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { ContributorCertificateResponse } from '../services/contribution-and-review-backend-api.service';
+import { ContributionAndReviewService } from '../services/contribution-and-review.service';
+
+interface CertificateContentData {
+  text: string;
+  linePosition: number;
+}
+
+@Component({
+  selector: 'certificate-download-modal',
+  templateUrl: './certificate-download-modal.component.html'
+})
+export class CertificateDownloadModalComponent {
+  @Input() suggestionType!: string;
+  @Input() username!: string;
+  @Input() languageCode!: string | null;
+  fromDate!: string;
+  toDate!: string;
+  errorMessage!: string;
+  errorsFound = false;
+  certificateDownloading = false;
+  datesSelected = false;
+
+  CERTIFICATE_WIDTH: number = 1500;
+  CERTIFICATE_HEIGHT: number = 1300;
+  CERTIFICATE_MID_POINT: number = 750;
+  CERTIFICATE_INNER_BOX_WIDTH: number = 1300;
+  CERTIFICATE_INNER_BOX_HEIGHT: number = 1100;
+  CANVAS_ROOT_COORDINATE: number = 0;
+  INNER_BOX_COORDINATE: number = 100;
+  LOGO_X_COORDINATE: number = 650;
+  LOGO_Y_COORDINATE: number = 150;
+  LOGO_WIDTH: number = 200;
+  LOGO_HEIGHT: number = 80;
+  SIGNATURE_BASE_COORDINATE: number = 400;
+  DATE_BASE_COORDINATE: number = 1100;
+  SIGNATURE_LINE_STARTING_POSITION: number = 300;
+  SIGNATURE_LINE_ENDING_POSITION: number = 500;
+  DATE_LINE_STARTING_POSITION: number = 1000;
+  DATE_LINE_ENDING_POSITION: number = 1200;
+
+  constructor(
+    private readonly activeModal: NgbActiveModal,
+    private contributionAndReviewService: ContributionAndReviewService) {
+  }
+
+  ngOnInit(): void {
+  }
+
+  close(): void {
+    this.activeModal.close();
+  }
+
+  downloadCertificate(): void {
+    this.errorsFound = false;
+    this.errorMessage = '';
+    if (
+      new Date(this.fromDate) >= new Date(this.toDate) ||
+      !this.fromDate ||
+      !this.toDate
```

Comment:
```suggestion
      !this.fromDate ||
      !this.toDate ||
      new Date(this.fromDate) >= new Date(this.toDate)
```

Should this be, like this ^

Reply:
Done